helix-core/src/main/java/org/apache/helix/ShardingKeyTrieNode.java

```diff
@@ -0,0 +1,66 @@
+package org.apache.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ShardingKeyTrieNode {
```

Comment:
In terms of defining APIs, please review the following pages:

https://docs.oracle.com/javase/7/docs/api/javax/swing/tree/TreeNode.html
https://fasterxml.github.io/jackson-core/javadoc/2.5/com/fasterxml/jackson/core/TreeNode.html

They list some commonly-used APIs for classes like this and you could also learn about what convention is used for naming the methods.

We are not going to need all the methods in these examples so let's start simple, but please try to follow the convention as that will help with reducing confusion.

Reply:
I agree to add common used APIs to this TrieNode class, like what Hunter already mentioned. The common used APIs will definitely help your development later.